PAC/common/device/device.h

```diff
@@ -1745,6 +1745,68 @@ class power_unit : public analog_io_device
 
         process_data_in p_data_in;
 
+        static process_data_out stub_p_data_out;
+        process_data_out* p_data_out = &stub_p_data_out;
+    };
+//-----------------------------------------------------------------------------
+/// @brief Конвертер IO-Link -> AO.
+///
+/// Устройство для преобразования IO-Link сигналов в аналоговые выходы.
+class converter_iolink_ao : public analog_io_device
+    {
+    public:
+        explicit converter_iolink_ao( const char* dev_name );
+
+        void direct_on() override;
+        void direct_off() override;
+
+        float get_value() override;
+        int get_state() override;
+
+        void direct_set_value( float val ) override;
+
+        void evaluate_io() override;
+
+        int save_device_ex( char* buff ) override;
+
+        int set_cmd( const char* prop, u_int idx, double val ) override;
+
+    private:
+        float v = .0f;  // Выходное значение.
+        int st = 0;     // Состояние устройства.
+        int err = 0;    // Ошибка.
+
+        enum CONSTANTS
+            {
+            C_AIAO_INDEX = 0,   ///< Индекс канала аналоговых данных.
+            };
+
+#pragma pack(push, 1)
+        struct process_data_in
+            {
+            uint16_t status_ch1 : 2;        // Статус канала 1.
+            uint16_t status_ch2 : 2;        // Статус канала 2.
+            uint16_t reserved   : 12;       // Резерв.
+            
+            uint16_t output_ch1;            // Выходное значение канала 1.
+            uint16_t output_ch2;            // Выходное значение канала 2.
+            };
```

Comment:
Fix structure according to this source - https://ioddfinder.io-link.com/productvariants/search/43435.